addOns/paramdigger/src/main/java/org/zaproxy/addon/paramdigger/ExtensionParamDigger.java

```diff
@@ -21,17 +21,20 @@
 
 import javax.swing.ImageIcon;
 import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.control.Control.Mode;
 import org.parosproxy.paros.extension.ExtensionAdaptor;
 import org.parosproxy.paros.extension.ExtensionHook;
 import org.parosproxy.paros.extension.ExtensionPopupMenuItem;
+import org.parosproxy.paros.extension.SessionChangedListener;
+import org.parosproxy.paros.model.Session;
 import org.parosproxy.paros.network.HttpMessage;
 import org.zaproxy.addon.paramdigger.gui.ParamDiggerDialog;
 import org.zaproxy.addon.paramdigger.gui.ParamDiggerPanel;
 import org.zaproxy.addon.paramdigger.gui.PopupMenuParamDigger;
 import org.zaproxy.zap.utils.DisplayUtils;
 import org.zaproxy.zap.view.ZapMenuItem;
 
-public class ExtensionParamDigger extends ExtensionAdaptor {
+public class ExtensionParamDigger extends ExtensionAdaptor implements SessionChangedListener {
```

Comment:
Should not be exposed through the extension.

Reply:
On the actual panel then?

Reply:
Here but an inner class.